src/page-modules/departures/layout.tsx

```diff
@@ -6,26 +6,49 @@ import Search from '@atb/components/search';
 import { Button } from '@atb/components/button';
 import style from './departures.module.css';
 import { useRouter } from 'next/router';
+import DepartureDateSelector, {
+  DepartureDateState,
+} from '@atb/components/departure-date-selector';
 
 export type DeparturesLayoutProps = PropsWithChildren<WithGlobalData<{}>>;
 function DeparturesLayout({ children }: DeparturesLayoutProps) {
   const { t } = useTranslation();
   const router = useRouter();
 
   const [selectedFeature, setSelectedFeature] = useState<GeocoderFeature>();
+  const [departureDateState, setDepartureDateState] = useState(
+    DepartureDateState.Now,
+  );
+  const [_departureDate, setDepartureDate] = useState<Date>();
```

Comment:
Would it be better with one state, a date, and that state gets updated accordingly inside the component changing the date and time separately?

Reply:
I think the API should be "cleaned" more here and communicate with timestamps instead. It is the only intended way of using it so we should do that data mapping / conversion as early as possible in the call stack

Reply:
So I think something like 

```tsx
type DepartureDate = {
  type: 'now'
} | {
  type: 'arrival',
  datetime: new Date()
} 
// | etc etc ;

<DepartureDateSelector
   initialValue={departureDate as DepartureDate}
    onChange={setDepartureDate}
  />

Reply:
Good suggestion! Was waiting with this some so that we agree on the solution. Implementing this post standup.

Reply:
When this is done I think it's ready to be merged for now